sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/context/AuthTokenInfo.java

```diff
@@ -49,6 +51,15 @@ public AuthTokenInfo(String authToken, Date startTime, Date endTime) {
         mEndTime = endTime;
     }
 
+    /**
+     * Returns true if token is about to expire.
+     *
+     * @return boolean, true if expires soon
+     */
+    public boolean isExpiresSoon() {
+        return (new Date().getTime() + EXPIRATION_OFFSET_TO_REFRESH_SEC * 1000) >= this.mEndTime.getTime();
```

Comment:
- Remove `this.`
- `mEndTime` can be `null`
- Use `before` or `after` methods instead compare numbers